---
title: "Effect of the diet type on the C. elegans transcriptome"
author: "Awan , Lotte, Nicky & Melanie"
date: "13 juni 2019"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction

The standard protocol for Caenorhabditis elegans growth and maintenance is 20 degrees Celsius on an Escherichia coli diet.During this analysis we focused on the changes in gene expression when the diet is changed to B.subtillis. 
For each condition (diet type) we obtained three replicates. With this script we determined the 10 most significantly expressed genes.


## Imports

Necessary libraries are loaded in first:
```{r, message=FALSE, warning=FALSE}
library(openxlsx  , quietly = T)
library(edgeR     , quietly = T)
library(ggplot2   , quietly = T)
library(data.table, quietly = T)
```


## Create countmatrix

Here we start with the count files for each sample. Each file contains the geneIDs and their corresponding total count. All samples and their count are put in one dataframe, counMatrix. Each column is named after diet type and the last two digits of the SRR code. 

```{r echo=TRUE}
  
#Open all count files
count85 <- (read.csv("counts_SRR5832185.txt", header = FALSE, sep = "\t"))
count86 <- (read.csv("counts_SRR5832186.txt", header = FALSE, sep = "\t"))
count87 <- (read.csv("counts_SRR5832187.txt", header = FALSE, sep = "\t"))

count94 <- (read.csv("counts_SRR5832194.txt", header = FALSE, sep = "\t"))
count95 <- (read.csv("counts_SRR5832195.txt", header = FALSE, sep = "\t"))
count96 <- (read.csv("counts_SRR5832196.txt", header = FALSE, sep = "\t"))


#split the chromosome and gene information
split1 <- do.call(rbind ,strsplit(as.character(count85[,1]), ":", fixed = TRUE))
split2 <- do.call(rbind ,strsplit(as.character(split1[,2]) , ".", fixed = TRUE))

#Add chromosome and gene information  to count85 dataframe and removes old column
count85 <- cbind(split2, count85)
count85[,"V1"] <- NULL


#creates count matrix 
countMatrix <<- do.call(cbind, list(count85, count86$V2, count87$V2, count94$V2, count95$V2, count96$V2))
colnames(countMatrix) <- c("Chromosome", "Gene", "E.coli_85", "E.coli_86", "E.coli_87", "B.subtillis_94", "B.subtillis_95", "B.subtillis_96")

#change rownames to GeneID
rownames(countMatrix) <- countMatrix[,"Gene"]

print(head(countMatrix))

```


## Low count filtering 

Genes with a low count are filtered out of the dataset (count < 50).

```{r}

# create DGE list
exp   <- c("E.coli", "E.coli", "E.coli", "B.subtillis", "B.subtillis", "B.subtillis")
group <- factor(exp)
y     <- DGEList(counts = countMatrix[,3:8], group = group)
  
# select all genes with at least 50 counts per million (cpm) in two samples
keep.genes <- rowSums(cpm(y) > 50) >= 2
y <- y[keep.genes,]
  
```


## Normalization

Normalization of the data was done using the TMM method.

```{r}
# recalculate the library size
y$samples$lib.size <- colSums(y$counts)

# normalization
y <- calcNormFactors(y, method = "TMM")
```


## Create Design Matrix

Design matrix is made to organize the samples and their corresponding conditions. 

```{r}
# create design matrix (samples grouped by conditions)
design <- model.matrix(~0+group, data = y$samples)
colnames(design) <- levels(y$samples$group)

print(head(design))
  
```


## Estimate dispersion

The common,Trended and Tagwise dispersion are calculated

```{r}
# estimate dispersion
y <- estimateGLMCommonDisp( y, design)
y <- estimateGLMTrendedDisp(y, design, method = "power")
y <- estimateGLMTagwiseDisp(y, design)
```


## Plot normalized data

First we plot the multidimensional scaling, which is a multivariate data analysis approach used for visualization of similarity and dissimilarity in a two dimensional plot. Second, we plot the BCV, which is the square root of the negative binomial dispersion. 

```{r}
# Plot MDS/PCA plot
plotMDS(y)

# Plot BCV plot
plotBCV(y)
```

## Create Dendogram

To check if the samples are clustered in the right groups a hierarchical clustering is performed. Distance matrix is made with euclidean distances and clustering is done with ward.D2 method. 

```{r}
# Create distance matrix of normalized count data
norm_counts <- as.data.frame(as.matrix.DGEList(y))
norm_counts <- na.omit(t(norm_counts))
norm_counts <- scale(norm_counts)
dist_mat <- dist(norm_counts, method = "euclidean")

# cluster data based on samples 
clustering <- hclust(dist_mat, method = "ward.D2")

# Plot hierarchical clustering
plot(clustering, ylab = "Distance", main = "Hierarchical Clustering of Samples")
```

## Determination of differentially expressed genes

All significantly expressed genes (p-value < 0.05) are filtered out and written to diff_expressed_genes.csv and data frame res. 

```{r}
# Determine differentially expressed genes
fit <- glmFit(y, design)
mc  <- makeContrasts(exp.r=E.coli-B.subtillis, levels = design)
fit <- glmLRT(fit, contrast = mc)
res <- topTags(fit, n = 100000, p.value = 0.05)
result_df <- as.data.frame(topTags(fit, n = 100000, p.value = 0.05))
  
write.csv(res ,"diff_expressed_genes.csv", row.names = TRUE)

print(head(result_df))
```

## Create input for future GSEA analysis

For a gene set enrichment analysis ad txt file with genes and their p-value and FDR are required. 

```{r}
subSet <- as.data.frame(res[,4:5])
setDT(subSet, keep.rownames = TRUE)[]

write.table(subSet, "gsea_input.txt", sep = "\t", row.names = FALSE)
```










